### ISSUE TITLE
CFD-413: White screen of death after deleting discussion

### ISSUE THREAD

**Issue by solker** [capacity4dev/capacity4more#576]
Scenario:
* Login with a normal user who is member of a group
* Go to a group the user is member of
* Open the quick post form to add a discussion, do not fill in any details
* Click on the "Open in Full form"

__Problem 1__ : An empty node is saved unpublished on the platform. 
A new activity is logged.

__Problem 2__ : Go to the Global home page
Unpublished node is in the activity stream > No unpublished nodes should be shown.

__Problem 3__ : Go to the Group Dashboard
Unpublished node is in the activity stream > No unpublished nodes should be shown.
 
Try to delete the empty node:
* Open the detail page, click edit
* Click on Delete
* Confirm

__Problem 4__ : User is redirected to the home page, a White Screen of death is shown.
Error in the watchdog:
```
EntityMetadataWrapperException: Unknown data property c4m_discussion_type. in EntityStructureWrapper->getPropertyInfo() (line 335 of /var/www/capacity4more-acc.dev.amplexor.com/capacity4more/capacity4more/modules/contrib/entity/includes/entity.wrapper.inc).
```

Quick fix for the WSOD : Delete the message record for the deleted node manually in the DB.

__The message should be deleted together with the node!__

**Comment by solker** [comment]
Problem 2 and Problem 3 fixed in #572
Problem 1 need to make re logic of program so we should discuss it. 
Problem 4 fixed in PR.

**Comment by nader77** [comment]
@solker Please move comments to Jira.
